server/aesdsocket.c

```diff
@@ -0,0 +1,394 @@
+    open ("/dev/null", O_RDWR);
+    dup (0); 
+    dup (0); 
+
+    return 0;
+}
+
+int main(int argc , char *argv[])
+{
+    openlog(NULL , 0 , LOG_USER);
+
+    // Register the signals 
+    signal(SIGINT , signalHandler);
+    signal(SIGTERM , signalHandler);
+
+    // Flag to track the if daemon is to be created
+    uint8_t daemon_flag = 0;
+
+    // Check if the -d argument was passed
+    char *daemon_arg = argv[1];
+    if(daemon_arg != NULL){
+        if (strcmp(daemon_arg , "-d") == 0){
+        daemon_flag = 1;
+        }
+        else{
+            printf("Please enter the correct argument '-d' for creating the daemon\n");
+            daemon_flag = 0;
+        }
+    }
+    
+
+    // Variable to store the return status
+    int ret_status = 0;
+    struct addrinfo *servinfo;
+    struct addrinfo hints;
+
+    // File to write the data from the socket
+    fd = open("/var/tmp/aesdsocketdata" , O_RDWR | O_CREAT | O_TRUNC, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+    // Error check
+    if (fd == -1){
+        syslog(LOG_ERR , "open\n");
+        return -1;
+    }
+    // printf("Value of fd:%d\n", fd);
+
+    // Variable to use for reusing the port logic
+    int check_port_status = 1;
+
+    // The structure has to empty initially
+    memset(&hints, 0, sizeof(hints));
+
+    // Set the attributes of the hints variable
+    hints.ai_family = AF_INET;     // IPv4 type of connection
+    hints.ai_socktype = SOCK_STREAM; // TCP stream sockets
+    hints.ai_flags = AI_PASSIVE;     // Fill in the IP
+ 
+
+    // Get the addrinfo structure for the port 9000
+    if ((ret_status = getaddrinfo(NULL, "9000", &hints, &servinfo)) != 0) {
+        syslog(LOG_ERR , "getaddrinfo\n");
+        return -1;
+    }
+
+    // Get the socket descriptor
+    sd = socket(servinfo->ai_family , servinfo->ai_socktype , servinfo->ai_protocol);
+    // Error check
+    if(sd == -1){
+        syslog(LOG_ERR , "socket\n");
+        return -1;
+    }
+
+    ret_status = setsockopt(sd , SOL_SOCKET , SO_REUSEADDR , &check_port_status , sizeof(check_port_status));
+    // Error check
+    if (ret_status == -1){
+        syslog(LOG_ERR , "setsocketopt\n");
+        return -1;
+    }
+
+    // Bind the socket to the local port
+    ret_status = bind(sd , servinfo->ai_addr , servinfo->ai_addrlen);
+    // Error check
+    if (ret_status == -1){
+        syslog(LOG_ERR , "bind\n");
+        return -1;
+    }
+
+    // Before listening for a new connection, after binding to port 9000, check if daemon is to be created
+    if (daemon_flag){
+        ret_status = createDaemon();
+
+        if (ret_status == -1){
+            syslog(LOG_ERR , "createDaemon\n");
+        }
+        else{
+            syslog(LOG_DEBUG , "Daemon created suceesfully\n");
+        }
+    }
+
+    ret_status = listen(sd , BACKLOG);
+    if (ret_status == -1){
+        perror("listen");
+        return -1;
+    }
+#ifdef DEBUG
```

Comment:
@KhyatiSatta remove unused debug code, use revision control to recall previous code if needed.